src/components/popup/popup.tsx

```diff
@@ -0,0 +1,150 @@
+        if (this.container) {
+            ReactDOM.unmountComponentAtNode(this.container);
+            this.container = null;
+        }
+    }
+
+    public render() {
+        this.setPosition();
+        this.popup = (
+            <div
+                data-automation-id="POPUP"
+                style={this.style}
+            >
+                {this.props.children}
+            </div>);
+        return null;
+    }
+
+    private setPosition = () => {
+        if (!this.props.anchor) {
+            return;
+        }
+        const newStyle = {...this.style};
+        newStyle.display = this.props.open ? '' : 'none';
+        const anchorRect = this.props.anchor.getBoundingClientRect();
+
+        newStyle.maxHeight = this.props.maxHeight;
+        newStyle.transform = '';
+        newStyle.WebkitTransform = '';
+        if (this.props.syncWidth) {
+            newStyle.width = anchorRect.width;
+        }
+
+        const anchorHorizontalPos = this.props.anchorPosition!.horizontal;
+        const anchorVerticalPos = this.props.anchorPosition!.vertical;
+
+        setTop(newStyle, anchorRect,
+            anchorVerticalPos, this.props.popupPosition!.vertical);
+        setLeft(newStyle, anchorRect,
+            anchorHorizontalPos, this.props.popupPosition!.horizontal);
+        if (this.props.syncWidth) {
+            newStyle.width = anchorRect.width;
+        }
+        this.style = newStyle;
+    }
+}
+
+function setTop(popupStyle: CSSProperties, anchorRect: ClientRect,
+                anchorPosition: VerticalPosition, popupPoistion: VerticalPosition) {
+    switch (popupPoistion) {
+        case 'top':
+            popupStyle.top = getVerticalReference(anchorRect, anchorPosition);
+            break;
+        case 'center':
+            popupStyle.top = getVerticalReference(anchorRect, anchorPosition);
+            addTransform(popupStyle, 'translateY(-50%)');
+            break;
+        case 'bottom':
+            popupStyle.top = getVerticalReference(anchorRect, anchorPosition);
+            addTransform(popupStyle, 'translateY(-100%)');
+            break;
+    }
+}
+
+function setLeft(popupStyle: CSSProperties, anchorRect: ClientRect,
+                 anchorPosition: HorizontalPosition, popupPoistion: HorizontalPosition) {
+    switch (popupPoistion) {
+        case 'left':
+            popupStyle.left = getHorizontalReference(anchorRect, anchorPosition);
+            break;
+        case 'center':
+            popupStyle.left = getHorizontalReference(anchorRect, anchorPosition);
+            addTransform(popupStyle, 'translateX(-50%)');
+            break;
+        case 'right':
+            popupStyle.left = getHorizontalReference(anchorRect, anchorPosition);
+            addTransform(popupStyle, 'translateX(-100%)');
+            break;
+    }
+}
+
+function getVerticalReference(rect: ClientRect, anchorPosition: VerticalPosition): number {
+    const yOffset = window.scrollY ? window.scrollY : window.pageYOffset;
```

Comment:
IE & Safari don't support scrollY (or partially support), I will change the statement to check if it's undefined.